src/ParallelAlgorithms/Interpolation/Callbacks/ObserveLineSegment.hpp

```diff
@@ -0,0 +1,114 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "DataStructures/Tensor/TensorData.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/Tags.hpp"
+#include "IO/Observer/VolumeActions.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/PostInterpolationCallback.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/PrettyType.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace intrp {
+namespace callbacks {
+
+/* \brief post_interpolation_callback that outputs requested tensors
+ * interpolated onto a LineSegment. The data is written as volume data into the
+ * `Reductions` file with Quadrature `CellCentered` and Basis
+ * `FiniteDifference`.
+ *
+ * Uses:
+ * - Metavariables
+ *   - `temporal_id`
+ * - DataBox:
+ *   - `TensorsToObserve`
+ * - GlobalCache:
+ *   - `observers::Tags::ReductionFileName`
+ *
+ * Conforms to the intrp::protocols::PostInterpolationCallback protocol
+ *
+ * For requirements on InterpolationTargetTag, see
+ * intrp::protocols::InterpolationTargetTag
+ *
+ */
+template <typename TensorsToObserve, typename InterpolationTargetTag>
+struct ObserveLineSegment
+    : tt::ConformsTo<intrp::protocols::PostInterpolationCallback> {
+  static constexpr double fill_invalid_points_with =
+      std::numeric_limits<double>::quiet_NaN();
+
+  using const_global_cache_tags =
+      tmpl::list<observers::Tags::ReductionFileName>;
+
+  template <typename DbTags, typename Metavariables, typename TemporalId>
+  static void apply(const db::DataBox<DbTags>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const TemporalId& temporal_id) {
+    static_assert(
+        tmpl::list_contains_v<TensorsToObserve,
+                              domain::Tags::Coordinates<
+                                  Metavariables::volume_dim, Frame::Inertial>>,
+        "The inertial coordinates must always be observed when observing "
+        "tensors on a line segment. This is so the data can be re-constructed "
+        "from the reductions file alone.");
```

Comment:
Maybe say `from the output file alone` to avoid explicitly saying reduction just to make it easier in the future when we rename things.